### PR TITLE
Ui/Download request replay

### PR DESCRIPTION
**Pull request by leon-up9**
image

### CODE REVIEW

File: ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx [hunk 1]
```diff
@@ -69,43 +67,63 @@ const decodeQueryParam = (p) => {
...
+
 const ReplayRequestModal: React.FC<ReplayRequestModalProps> = ({ isOpen, onClose }) => {
     const entryData = useRecoilValue(entryDataAtom)
     const request = entryData.data.request
-    const [method, setMethod] = useState(request?.method?.toLowerCase() as string)
     const getHostUrl = useCallback(() => {
         return entryData.data.dst.name ? entryData.data?.dst?.name : entryData.data.dst.ip
     }, [entryData.data.dst.ip, entryData.data.dst.name])
-    const [hostPortInput, setHostPortInput] = useState(`${entryData.base.proto.name}://${getHostUrl()}:${entryData.data.dst.port}`)
+    const getHostPortVal = useCallback(() => {
```
Comment (lirazyehezkel) [inline ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx:81]: can you move all the useCallbacks to be under all the states?

**Review by lirazyehezkel — COMMENTED** [review]
small comments

File: ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx (hunk 1 above)
Comment (leon-up9) [reply ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx:81]: nope, cause it has to be declared before used

File: ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx [hunk 2]
```diff
@@ -69,43 +67,63 @@ const decodeQueryParam = (p) => {
...
     useEffect(() => {
-        const newParams = getQueryStringParams(debouncedPath);
-        setParams(convertParamsToArr(newParams))
+        onParamsChange(requestDataModel.params)
+    }, [onParamsChange, requestDataModel.params])
+
+    useEffect(() => {
+        const params = convertParamsToArr(getQueryStringParams(debouncedPath));
+        setRequestData({ ...requestDataModel, params })
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (leon-up9) [reply ui-common/src/components/modals/ReplayRequestModal/ReplayRequestModal.tsx:126]: i couldnt find a way

**Review by lirazyehezkel — APPROVED** [review]

**Review by lirazyehezkel — APPROVED** [review]

**Pull request merged**